source/org/zfin/feature/repository/HibernateFeatureRepository.java

```diff
@@ -501,41 +507,28 @@ public String getPrefixById(int labPrefixID) {
     public String getCurrentPrefixForLab(String labZdbID) {
         Session session = HibernateUtil.currentSession();
         String hqlLab1 = " select fp.prefixString from  OrganizationFeaturePrefix  lfp join lfp.organization l " +
-                " join lfp.featurePrefix fp " +
-                " where l.zdbID =:labZdbID" +
-                " and lfp.currentDesignation =:currentDesignation ";
-        Query queryLab = session.createQuery(hqlLab1);
-        queryLab.setString("labZdbID", labZdbID);
-        queryLab.setBoolean("currentDesignation", true);
+                         " join lfp.featurePrefix fp " +
+                         " where l.zdbID =:labZdbID" +
+                         " and lfp.currentDesignation =:currentDesignation ";
+        Query<String> queryLab = session.createQuery(hqlLab1, String.class);
+        queryLab.setParameter("labZdbID", labZdbID);
+        queryLab.setParameter("currentDesignation", true);
         return (String) queryLab.uniqueResult();
 
     }
 
 
     public List<Organization> getLabsOfOriginWithPrefix() {
-        String hqlLab = " select distinct lfp.organization from OrganizationFeaturePrefix lfp  " +
-                " where lfp.featurePrefix is not null " +
-                " and lfp.organization.name is not null  " +
-                " and lfp.currentDesignation = :true";
-
-        Query query = HibernateUtil.currentSession().createQuery(hqlLab);
-        query.setBoolean("true", true);
-        /*List<FeatureMarkerRelationship> featureMarkerRelationships = (List<FeatureMarkerRelationship>) query.list();
-              HibernateUtil.currentSession().createQuery(hqlLab)
-                .setBoolean("true", true)*/
-                /*setResultTransformer(new BasicTransformerAdapter() {
-                    @Override
-                    public Object transformTuple(Object[] tuple, String[] aliases) {
-                        return tuple[0];
-                    }
-                })*/
-        List<Organization> organizations = (List<Organization>) query.list();
-        Collections.sort(organizations, new Comparator<Organization>() {
-            @Override
-            public int compare(Organization o1, Organization o2) {
-                return o1.getName().compareToIgnoreCase(o2.getName());
-            }
-        });
+        String hqlLab = """ 
+            select distinct lfp.organization from OrganizationFeaturePrefix lfp
+            where lfp.featurePrefix is not null
+            and lfp.organization.name is not null
+            and lfp.currentDesignation = :true
+            """;
+        Query<Organization> query = HibernateUtil.currentSession().createQuery(hqlLab, Organization.class);
+        query.setParameter("true", true);
```

Comment:
This is a minor thing, but I usually remove parameters like this and just update the hql to be "currentDesignation = true"

Reply:
Sure.